### PR TITLE
gh-534 - Deprecate unnecessary operations

### PR DESCRIPTION
**Pull request by t616178**
Work in progress.

### CODE REVIEW

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetEdgesBySeedTest.java [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import uk.gov.gchq.gaffer.exception.SerialisationException;
 import uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser;
 import uk.gov.gchq.gaffer.operation.GetOperation;
+import uk.gov.gchq.gaffer.operation.GetOperation.SeedMatchingType;
 import uk.gov.gchq.gaffer.operation.OperationTest;
 import uk.gov.gchq.gaffer.operation.data.EdgeSeed;
 import java.util.Arrays;
```
Comment (p013570) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetEdgesBySeedTest.java]: Please could we move these tests into GetEdgesTest.java.

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetElementsBySeedTest.java [hunk 2]
```diff
@@ -17,11 +17,11 @@
 package uk.gov.gchq.gaffer.operation.impl.get;
```
Comment (p013570) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetElementsBySeedTest.java]: Please could we move these tests into GetElementsTest.java

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetEntitiesBySeedTest.java [hunk 3]
```diff
@@ -21,6 +21,7 @@
 import uk.gov.gchq.gaffer.exception.SerialisationException;
 import uk.gov.gchq.gaffer.jsonserialisation.JSONSerialiser;
 import uk.gov.gchq.gaffer.operation.GetOperation;
+import uk.gov.gchq.gaffer.operation.GetOperation.SeedMatchingType;
```
Comment (p013570) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetEntitiesBySeedTest.java]: Please could we move these tests into GetEntitiesTest.java

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedEdgesTest.java [hunk 4]
```diff
@@ -27,9 +27,9 @@
 import java.util.Collections;
 import java.util.Iterator;
 
+import static junit.framework.TestCase.assertFalse;
```
Comment (p013570) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedEdgesTest.java]: Please could we move these tests into GetEdgesTest.java

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedElementsTest.java [hunk 5]
```diff
@@ -41,7 +41,7 @@ public void shouldSetSeedMatchingTypeToRelated() {
         final ElementSeed elementSeed2 = new EdgeSeed("source2", "destination2", true);
 
         // When
-        final GetRelatedElements op = new GetRelatedElements(Arrays.asList(elementSeed1, elementSeed2));
+        final GetElements op = new GetElements(Arrays.asList(elementSeed1, elementSeed2));
```
Comment (p013570) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedElementsTest.java]: Please could we move these tests into GetElementsTest.java

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedEntitiesTest.java [hunk 6]
```diff
@@ -42,7 +42,7 @@ public void shouldSetSeedMatchingTypeToRelated() {
         final EdgeSeed seed1 = new EdgeSeed("source1", "destination1", true);
 
         // When
-        final GetRelatedEntities<EdgeSeed> op = new GetRelatedEntities<>(Collections.singletonList(seed1));
+        final GetEntities op = new GetEntities(Collections.singletonList(seed1));
```
Comment (p013570) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedEntitiesTest.java]: Please could we move these tests into GetEntitiesTest.java

File: core/store/src/test/java/uk/gov/gchq/gaffer/store/StoreTest.java [hunk 7]
```diff
@@ -307,15 +296,16 @@ public void shouldHandleMultiStepOperations() throws Exception {
...
+        final GetElements<ElementSeed, Element> getElementsBySeed = new GetElements<>();
         final OperationChain<CloseableIterable<Element>> opChain = new OperationChain.Builder()
                 .first(addElements1)
                 .then(getElementsBySeed)
                 .build();
 
 
         given(addElementsHandler.doOperation(addElements1, context, store)).willReturn(null);
-        given(getElementsHandler.doOperation(getElementsBySeed, context, store)).willReturn(getElementsResult);
+        given(getElementsHandler.doOperation(getElementsBySeed, context, store))
```
Comment (p013570) [inline core/store/src/test/java/uk/gov/gchq/gaffer/store/StoreTest.java]: Perhaps do a quick global search for getElementsBySeed and replace it with getElements.

File: example/example-graph/src/main/java/uk/gov/gchq/gaffer/example/operation/GetElementsBySeedExample.java [hunk 8]
```diff
@@ -24,15 +24,15 @@
 import uk.gov.gchq.gaffer.operation.data.EdgeSeed;
 import uk.gov.gchq.gaffer.operation.data.ElementSeed;
 import uk.gov.gchq.gaffer.operation.data.EntitySeed;
-import uk.gov.gchq.gaffer.operation.impl.get.GetElementsBySeed;
+import uk.gov.gchq.gaffer.operation.impl.get.GetElements;
 
 public class GetElementsBySeedExample extends OperationExample {
```
Comment (p013570) [inline example/example-graph/src/main/java/uk/gov/gchq/gaffer/example/operation/GetElementsBySeedExample.java]: Perhaps we could merge GetElementsBySeedExample and GetRelatedElementsExample into GetElementsExample. Similarly with Entities and Edges.

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetEdgesBySeedTest.java (hunk 1 above)
Comment (t616178) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetEdgesBySeedTest.java]: Done

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedEdgesTest.java (hunk 4 above)
Comment (t616178) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedEdgesTest.java]: Done

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedEntitiesTest.java (hunk 6 above)
Comment (t616178) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedEntitiesTest.java]: Done

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetEntitiesBySeedTest.java (hunk 3 above)
Comment (t616178) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetEntitiesBySeedTest.java]: Done

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetElementsBySeedTest.java (hunk 2 above)
Comment (t616178) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetElementsBySeedTest.java]: Done

File: core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedElementsTest.java (hunk 5 above)
Comment (t616178) [inline core/operation/src/test/java/uk/gov/gchq/gaffer/operation/impl/get/GetRelatedElementsTest.java]: Done

**Pull request merged**